### DISCUSSION TITLE
Mistake in docs regarding `run_once_` vs `run_onchange_`?

### DISCUSSION THREAD

**Discussion by triallax** [twpayne/chezmoi#2744] [Q&A] [answered]
[This section of the docs](https://www.chezmoi.io/user-guide/use-scripts-to-perform-actions/#understand-how-scripts-work) states:

> Scripts that should only be run if they have not been run before have the prefix run_once_. Scripts that should be run whenever their contents change have the run_onchange_ prefix.

However, [another section in the same docs page](https://www.chezmoi.io/user-guide/use-scripts-to-perform-actions/#install-packages-with-scripts) states:

> As it [the `run_once_install-packages.sh` script] has the run_once_ prefix, it will not be run again unless its contents change, for example if you add more packages to be installed.

These statements are in direct contradiction, so which one is true? Or am I missing something?

**Comment by twpayne** [comment, marked as answer]
The first one is correct. Thanks for spotting this.

**Reply by twpayne** [reply]
https://github.com/twpayne/chezmoi/pull/2745 fixes the documentation.

The discrepancy occurred because initially chezmoi only had `run_` and `run_once_` scripts. `run_onchange_` scripts were added later, and although most of the documentation was updated, this was clearly missed.

**Reply by triallax** [reply]
Thank you!
